tests/Functional/ViewHelpers/LengthViewHelperTest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file belongs to the package "TYPO3 Fluid".
+ * See LICENSE.txt that was shipped with this package.
+ */
+
+namespace TYPO3\CMS\Fluid\Tests\Functional\ViewHelpers;
+
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\Attributes\Test;
+use TYPO3Fluid\Fluid\Tests\Functional\AbstractFunctionalTestCase;
+use TYPO3Fluid\Fluid\View\TemplateView;
+
+final class LengthViewHelperTest extends AbstractFunctionalTestCase
+{
+    public static function renderValidDataProvider(): iterable
+    {
+        yield 'single element' => [
+            'arguments' => [
+                'value' => 'Hello, World!',
+            ],
+            'src' => '<f:length value="{value}" />',
+            'expectation' => 13,
+        ];
+        yield 'single element, with encoding' => [
+            'arguments' => [
+                'value' => 'Hello, World!',
+                'encoding' => 'UTF-8',
+            ],
+            'src' => '<f:length value="{value}" encoding="{encoding}" />',
+            'expectation' => 13,
+        ];
+        yield 'single element inline' => [
+            'arguments' => [
+                'value' => 'Hello, World!',
+            ],
+            'src' => '{f:length(value: value)}',
+            'expectation' => 13,
+        ];
+        yield 'single element inline, pipe variable' => [
+            'arguments' => [
+                'value' => 'Hello, World!',
+            ],
+            'src' => '{value -> f:length()}',
+            'expectation' => 13,
+        ];
+    }
+
+    #[DataProvider('renderValidDataProvider')]
+    #[Test]
+    public function renderValid(array $arguments, string $src, mixed $expectation): void
+    {
+        $view = new TemplateView();
+        $view->getRenderingContext()->setCache(self::$cache);
+        $view->getRenderingContext()->getTemplatePaths()->setTemplateSource($src);
+        $view->assignMultiple($arguments);
+        self::assertSame($expectation, $view->render());
+
+        $view = new TemplateView();
+        $view->getRenderingContext()->setCache(self::$cache);
+        $view->getRenderingContext()->getTemplatePaths()->setTemplateSource($src);
+        $view->assignMultiple($arguments);
+        self::assertSame($expectation, $view->render());
+    }
+
+    public static function renderInvalidDataProvider(): iterable
+    {
+        yield 'invalid value' => [
+            'arguments' => [
+                'value' => 13,
```

Comment:
I would argue that this should be a valid use case:

```
<f:length value="13" />

Result: 2
```

Maybe use `is_scalar()` in your code above and then cast to string?